eating_cookies/eating_cookies.py

```diff
@@ -5,8 +5,14 @@
 # The cache parameter is here for if you want to implement
 # a solution that is more efficient than the naive 
 # recursive solution
-def eating_cookies(n, cache=None):
-  pass
+# Can eat 3, 2, or 1 cookie(s) at a time
+
+def eating_cookies(n, cache={0: 1, 1: 1, 2: 2, 3: 4, 4: 7}):
```

Comment:
the idea with the cache is that you don't actually fill it out manually beforehand haha. It gets filled out as you run the function so answers that were stored already don't have to be found again, if that makes sense.

the solution code for this one is:

if n < 0:
____return 0
elif n == 0:
____return 1
elif cache and cache[n] > 0:
____return cache[n]
else:
____if not cache:
________cache = {i: 0 for i in range(n+1)}
____cache[n] = eating_cookies(n-1, cache) + eating_cookies(n-2, cache) + eating_cookies(n-3, cache)
____return cache[n]